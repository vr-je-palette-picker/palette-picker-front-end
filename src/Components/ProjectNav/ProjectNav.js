import React, { Component } from 'react';
import { createNewProject } from '../../utils/apiCalls/apiCalls.js'
import './ProjectNav.scss';

export class ProjectNav extends Component {
  constructor() {
    super();
    this.state = {
      input: ''
    }
  }
  // createProjectSelectOptions = () => {
  //   const { projects } = this.props;
  //   let options = projects.map((project, index) => {
  //     return <option key={index} value={project.project_name}></option>;
  //   });
  //   return options;
  // };

  handleChange = (e) => {
    this.setState({input: e.target.value});
  };

  createNew = () => {
    createNewProject({project_name: this.state.input});
  };

  render() {
    return (
      <div className='ProjectNav'>
        <h1 className='ProjectNav__h1--title'>Palette Picker</h1>
        <input placeholder='Enter Project Name' id='new-project-input' onChange={(e) => this.handleChange(e)}/>
        <p className='ProjectNav__button--new-project' onClick={() => this.createNew()}>Add New Project</p>
        {/* <select className='ProjectNav__select--project'>
          {this.createProjectSelectOptions()}
        </select> */}
      </div>
    );
  }
}

export default ProjectNav;
