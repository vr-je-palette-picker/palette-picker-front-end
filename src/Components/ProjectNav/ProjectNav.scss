.ProjectNav {
  display: flex;
  flex-direction: row;
  height: 15vh;
  justify-content: space-between;
  min-height: 70px;
  width: 100vw;
    h1 {
      margin-left: 20px;
    }
}

.ProjectNav__div--add-container {
  align-self: flex-end;
  display: flex;
  height: 20%;
    input {
      height: 100%;
      margin-right: 10px;
    }
    p {
      cursor: pointer;
      margin: 0;
    }
  margin-right: 20px;

}
