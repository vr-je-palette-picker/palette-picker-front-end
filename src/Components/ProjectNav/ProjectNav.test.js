import React from 'react';
import { shallow } from 'enzyme';
import ProjectNav from './ProjectNav';

describe('ProjectNav', () => {
  let wrapper;
  let projects = [{}, {}, {}];

  beforeEach(() => {
    wrapper = shallow(<ProjectNav projects={projects} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when handleChange is called', () => {
    const mockEvent = {
      target: {
        name: 'input',
        value: 'Living Room Remodel'
      }
    };
    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state()).toEqual({ input: 'Living Room Remodel' })
  });

  it('should call handleChange onChange', () => {
    wrapper.instance().handleChange = jest.fn();
    wrapper.find('input').simulate('change')

    expect(wrapper.instance().handleChange).toHaveBeenCalled();
  })

  it('should call checkUnique onClick', () => {
    wrapper.instance().checkUnique = jest.fn();
    wrapper.find('p').simulate('click');

    expect(wrapper.instance().checkUnique).toHaveBeenCalled();
  })
});
