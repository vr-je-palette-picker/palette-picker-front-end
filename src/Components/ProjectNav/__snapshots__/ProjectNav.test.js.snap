// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectNav should match the snapshot 1`] = `
<div
  className="ProjectNav"
>
  <h1
    className="ProjectNav__h1--title"
  >
    Palette Projects
  </h1>
  <div
    className="ProjectNav__div--add-container"
  >
    <input
      id="new-project-input"
      onChange={[Function]}
      placeholder="Enter Project Name"
      value=""
    />
    <p
      className="ProjectNav__button--new-project"
      onClick={[Function]}
    >
      Add New Project
    </p>
  </div>
</div>
`;
